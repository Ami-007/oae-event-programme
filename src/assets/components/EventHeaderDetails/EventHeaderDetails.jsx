import "./EventHeaderDetails.scss";

const EventHeaderDetails = (props) => {
  const { time, date, venue, location } = props;
  return (
    <div className="event-details">
      <p className="event-details__time-date">
        {time}
        <span className="event-details__divider">|</span>
        {date}
      </p>
      <p className="event-details__venue">
        {venue}
        <span className="event-details__location">, {location}</span>
      </p>
    </div>
  );
};

export default EventHeaderDetails;
