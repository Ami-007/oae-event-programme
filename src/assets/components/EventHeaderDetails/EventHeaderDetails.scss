@use "../../sass/variables" as *;
@use "../../sass/typography" as *;

.event-details {
  color: white;
  background-color: #fa825d;
  line-height: 24px;
  & > * {
    margin: 0;
  }
  &__divider {
    margin-left: 5px;
    margin-right: 5px;
  }
  &__venue {
    font-weight: bold;
  }
  &__location {
    font-weight: normal;
  }
}
