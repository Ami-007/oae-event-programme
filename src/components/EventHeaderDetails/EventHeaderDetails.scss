@use "../../assets/sass/variables.scss" as *;
@use "../../assets/sass/typography.scss" as *;

.event-header-details {
  color: $color-white;
  background-color: $modern-primary;
  line-height: 24px;
  font-family: $modern-content;
  padding: 23px;
  & > * {
    margin: 0;
  }
  &__divider {
    margin-left: 5px;
    margin-right: 5px;
  }
  &__venue {
    font-weight: bold;
  }
  &__location {
    font-weight: normal;
  }
}
