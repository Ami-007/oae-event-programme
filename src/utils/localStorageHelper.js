export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
