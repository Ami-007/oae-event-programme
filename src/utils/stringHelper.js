export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
