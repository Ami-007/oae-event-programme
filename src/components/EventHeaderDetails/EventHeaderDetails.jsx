import "./EventHeaderDetails.scss";

const EventHeaderDetails = (props) => {
  const { time, date, venue, location } = props;
  return (
    <div className="event-header-details">
      <p className="event-header-details__time-date">
        {time}
        <span className="event-header-details__divider">|</span>
        {date}
      </p>
      <p className="event-header-details__venue">
        {venue}
        <span className="event-header-details__location">, {location}</span>
      </p>
    </div>
  );
};

export default EventHeaderDetails;
