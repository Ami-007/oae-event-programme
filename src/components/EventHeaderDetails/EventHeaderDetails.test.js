import EventHeaderDetails from "./EventHeaderDetails";
import { customRender } from "../../utils/testUtils";

test(" EventHeaderDetails component", () => {
  const { container } = customRender(<EventHeaderDetails />);
  expect(container).toMatchSnapshot();
});

test("time should display as 11.30am", () => {
  const { container } = customRender(<EventHeaderDetails time="11.30am" />);
  expect(container).toMatchSnapshot();
});

test("date should be Sun 21 Nov 2021", () => {
  const { container } = customRender(<EventHeaderDetails date="Sun 21 Nov 2021" />);
  expect(container).toMatchSnapshot();
});

test("venue should be King's place", () => {
  const { container } = customRender(<EventHeaderDetails venue="King's place" />);
  expect(container).toMatchSnapshot();
});

test("location should be London", () => {
  const { container } = customRender(<EventHeaderDetails location="London" />);
  expect(container).toMatchSnapshot();
});

test("Vertical divider span element should only render when there is both a time and date", () => {
  const { container } = customRender(<EventHeaderDetails time="11.30am" date="Sun 21 Nov 2021" venue="London South Bank" location="London" />);
  expect(container).toMatchSnapshot();
});

test("Comma divider span element should not render if there is no location", () => {
  const { container } = customRender(<EventHeaderDetails time="11.30am" date="Sun 21 Nov 2021" venue="London South Bank" />);
  expect(container).toMatchSnapshot();
});

test("Comma divider span element should render if there is a venue and location", () => {
  const { container } = customRender(<EventHeaderDetails time="11.30am" date="Sun 21 Nov 2021" venue="London South Bank" location="London" />);
  expect(container).toMatchSnapshot();
});
