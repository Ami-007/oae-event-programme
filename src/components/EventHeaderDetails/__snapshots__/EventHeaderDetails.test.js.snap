// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` EventHeaderDetails component 1`] = `
<div>
  <div
    class="event-header-details"
  >
    <p
      class="event-header-details__time-date"
    />
    <p
      class="event-header-details__venue-location"
    >
      <span
        class="event-header-details__venue"
      />
       
    </p>
  </div>
</div>
`;

exports[`Comma divider span element should not render if there is no location 1`] = `
<div>
  <div
    class="event-header-details"
  >
    <p
      class="event-header-details__time-date"
    >
      11.30am
      <span
        class="event-header-details__divider"
      >
        |
      </span>
      Sun 21 Nov 2021
    </p>
    <p
      class="event-header-details__venue-location"
    >
      <span
        class="event-header-details__venue"
      >
        London South Bank
      </span>
       
    </p>
  </div>
</div>
`;

exports[`Comma divider span element should render if there is a venue and location 1`] = `
<div>
  <div
    class="event-header-details"
  >
    <p
      class="event-header-details__time-date"
    >
      11.30am
      <span
        class="event-header-details__divider"
      >
        |
      </span>
      Sun 21 Nov 2021
    </p>
    <p
      class="event-header-details__venue-location"
    >
      <span
        class="event-header-details__venue"
      >
        London South Bank
      </span>
      <span
        class="event-header-details__comma"
      >
        ,
      </span>
       
      London
    </p>
  </div>
</div>
`;

exports[`Vertical divider span element should only render when there is both a time and date 1`] = `
<div>
  <div
    class="event-header-details"
  >
    <p
      class="event-header-details__time-date"
    >
      11.30am
      <span
        class="event-header-details__divider"
      >
        |
      </span>
      Sun 21 Nov 2021
    </p>
    <p
      class="event-header-details__venue-location"
    >
      <span
        class="event-header-details__venue"
      >
        London South Bank
      </span>
      <span
        class="event-header-details__comma"
      >
        ,
      </span>
       
      London
    </p>
  </div>
</div>
`;

exports[`date should be Sun 21 Nov 2021 1`] = `
<div>
  <div
    class="event-header-details"
  >
    <p
      class="event-header-details__time-date"
    >
      Sun 21 Nov 2021
    </p>
    <p
      class="event-header-details__venue-location"
    >
      <span
        class="event-header-details__venue"
      />
       
    </p>
  </div>
</div>
`;

exports[`location should be London 1`] = `
<div>
  <div
    class="event-header-details"
  >
    <p
      class="event-header-details__time-date"
    />
    <p
      class="event-header-details__venue-location"
    >
      <span
        class="event-header-details__venue"
      />
      <span
        class="event-header-details__comma"
      >
        ,
      </span>
       
      London
    </p>
  </div>
</div>
`;

exports[`time should display as 11.30am 1`] = `
<div>
  <div
    class="event-header-details"
  >
    <p
      class="event-header-details__time-date"
    >
      11.30am
    </p>
    <p
      class="event-header-details__venue-location"
    >
      <span
        class="event-header-details__venue"
      />
       
    </p>
  </div>
</div>
`;

exports[`venue should be King's place 1`] = `
<div>
  <div
    class="event-header-details"
  >
    <p
      class="event-header-details__time-date"
    />
    <p
      class="event-header-details__venue-location"
    >
      <span
        class="event-header-details__venue"
      >
        King's place
      </span>
       
    </p>
  </div>
</div>
`;
